src/buffer/out/cursor.cpp

```diff
@@ -13,7 +13,6 @@
 // - ulSize - The height of the cursor within this buffer
 Cursor::Cursor(const ULONG ulSize, TextBuffer& parentBuffer) noexcept :
     _parentBuffer{ parentBuffer },
-    _cPosition{ 0 },

```

Comment:
Are we sure this gets 0 initialized always? I know we used to have some weird OS-side bugs where memory wouldn't get 0 initialized. (apologies for needing a refresher on basic language concepts)

Reply:
To expand on that:
The old `COORD _cPosition` needed a manual initialization unlike `til::point`, because that type has no custom constructor which initializes `X`/`Y` to 0, nor does it use C++11's default member initializers (`short X = 0;`) like `til::point` does, so the default value for its members is "uninitialized" (which is UB).

The reason the existing code said `_cPosition{ 0 }` and not `_cPosition{}` is probably a carry-over from the C code, because C structs don't have default constructors - `_cPosition{}` is not valid C.
In C, if you pass less arguments than the struct has members, then the last argument will be assigned to all remaining fields. `RECT{ 0 }` is equivalent to `RECT{ 0, 0, 0, 0 }` and `RECT{ 0, 1 }` is equivalent to `RECT{0, 1, 1, 1 }`.
In C++, the compiler-generated constructor acts equivalent to C's with the addition of the default constructor (`{}`) which (practically speaking) assigns 0 to all such fields.

Since `til::point` has a custom constructor it gets no compiler-generated one and thus it can't be used with C's `{ 0 }` style constructor arguments anymore.